agent/src/main/scala/za/co/absa/atum/agent/dispatcher/Dispatcher.scala

```diff
@@ -18,8 +18,21 @@ package za.co.absa.atum.agent.dispatcher
 
 import za.co.absa.atum.model.dto.{AtumContextDTO, CheckpointDTO, PartitioningSubmitDTO}
 
+/**
+ *  This trait provides a contract for different dispatchers
+ */
 trait Dispatcher {
+  /**
+   *  This method is used to ensure the server knows the given partitioning. 
+   *  As a response the `AtumContext` is fetched from the server.
+   *  @param partitioning: PartitioningSubmitDTO to be used to ensure server knows the given partitioning.
+   *  @return AtumContextDTO.
+   */
   def createPartitioning(partitioning: PartitioningSubmitDTO): AtumContextDTO
 
+  /**
+   *  This method is used to save checkpoint to server.
```

Comment:
or, to data store? Or, to send to server sfor storing it into data store? Or we don't want to be that precise, since it's a general dispatcher?

Reply:
Actually no, not all dispatchers are/will be working with data stores

Reply:
I guess this depends on how we resolve the one above.

Reply:
yeah let's keep it as is then